packages/dcos-integration-test/extra/test_packaging.py

```diff
@@ -0,0 +1,151 @@
+    return r.json()
+
+
+def _agent_has_resources(agent, node_requirements):
+    """Check that an agent has at least as much resources as requried for one node
+
+    Args:
+        agent: dict Info for one 'slave' from the mesos state summary
+        node_requirements: dict Resource requirements per agent
+    """
+    unreserved = agent['unreserved_resources']
+    resources = ['mem', 'disk', 'cpus']
+    for resource in resources:
+        log.debug('{resource}: unreserved {unreserved}, required {required}'.format(
+            resource=resource,
+            unreserved=unreserved[resource],
+            required=node_requirements[resource]))
+        if unreserved[resource] < node_requirements[resource]:
+            log.debug('Agent does not have has enough {}'.format(resource))
+            return False
+    return True
+
+
+def _enough_resources_for_package(state_summary, package_requirements):
+    """Sanity check that there are enough unreserved_resources in a state to run a package.
+
+    Args:
+        state_summary: dict
+        package_requirements: dict
+
+    Returns:
+        bool
+
+    Notes:
+        This is a sanity check meant for use with pytest.mark.skipif
+        This only verifies that there were enough resources at the time state_summary was queried
+    """
+    agents = state_summary['slaves']
+    if len(agents) < package_requirements['number_of_nodes']:
+        log.debug('Not enough agents for this package. Need {required}, have {available}'.format(
+            required=package_requirements['number_of_nodes'],
+            available=len(agents)))
+        return False
+    usable_nodes = 0
+    for agent in agents:
+        if _agent_has_resources(agent, package_requirements['node']):
+            usable_nodes += 1
+        if usable_nodes == package_requirements['number_of_nodes']:
+            return True
+    log.info('Only {available} usable agents. This package needs {required} with these resources {resources}.'.format(
+        available=usable_nodes,
+        required=package_requirements['number_of_nodes'],
+        resources=package_requirements['node']))
+    return False
+
+
+def _skipif_insufficient_resources(dcos_api_session, requirements):
+    """Can't access dcos_api_session from through the pytest.mark.skipif decorator, so call this in each test instead
+    """
+    if not _enough_resources_for_package(_get_cluster_resources(dcos_api_session), requirements):
+        return pytest.skip(msg='Package installation would fail on this cluster due to insufficient resources')
+
+
+def test_packaging_api(dcos_api_session):
+    """Test the Cosmos API (/package) wrapper
+    """
+    _skipif_insufficient_resources(dcos_api_session, KAFKA_PACKAGE_REQUIREMENTS)
+    install_response = dcos_api_session.cosmos.install_package('kafka', '1.1.9-0.10.0.0')
+    data = install_response.json()
+
+    dcos_api_session.marathon.poll_marathon_for_app_deployment(data['appId'], 1,
```

Comment:
This is inconsequential, but instead of polling for this specific application you can just broadly wait for all deployments to end with this: https://github.com/dcos/dcos-test-utils/blob/master/dcos_test_utils/marathon.py#L298 and then have a much simpler function signature